src/eviction/eviction.c

```diff
@@ -87,20 +91,35 @@ static inline uint32_t ocf_evict_do(ocf_cache_t cache,
 			/* It seams that no more partition for eviction */
 			break;
 		}
-		if (evicted >= evict_cline_no) {
-			/* Evicted requested number of cache line, stop */
-			goto out;
-		}
 
-		to_evict = ocf_evict_calculate(cache, part, evict_cline_no,
-				true);
+		/* in first loop iteration only consider overflown partitions */
+		overflow_size = ocf_part_overflow_size(cache, part);
+		if (overflown && overflow_size == 0)
+			continue;
+
+		to_evict = ocf_evict_calculate(cache, part,
+				evict_cline_no - evicted, true);
```

Comment:
note third argument is changed from evict_cline_no to 'evict_cline_no - evicted'